clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/client/NettyFileTransferClient.java

```diff
@@ -373,18 +373,42 @@ private FullHttpRequest prepareRequest(
   }
 
   /**
-   * Connects to the host
+   * Connects to the host with a timeout
    */
   private Channel connectToHost(String host, String storeName, int version, int partition) {
+    ChannelFuture connectFuture = null;
+    String replicaId = Utils.getReplicaId(Version.composeKafkaTopic(storeName, version), partition);
     try {
-      return clientBootstrap.connect(host, serverPort).sync().channel();
+      connectFuture = clientBootstrap.connect(host, serverPort);
+
+      // Wait at most CONNECTION_ESTABLISHMENT_TIMEOUT_MS for the connect to finish
+      boolean completed =
+          connectFuture.awaitUninterruptibly(CONNECTION_ESTABLISHMENT_TIMEOUT_MS, TimeUnit.MILLISECONDS);
```

Comment:
Hmm, I believe this API still blocks on the current call thread. So essentially you connection will still timeout because from what we discussed, it is waiting for the channel registration which needs this thread to execute....